src/Emulator/Peripherals/Peripherals/Timers/STM32F4_Watchdog.cs

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2021 Zisis Adamos
+//
+//  This file is licensed under the MIT License.
+//  Full license text is available in 'licenses/MIT.txt'.
+//
+
+using System;
+using Antmicro.Renode.Core;
+using Antmicro.Renode.Core.Structure.Registers;
+using Antmicro.Renode.Time;
+using Antmicro.Renode.Logging;
+
+namespace Antmicro.Renode.Peripherals.Timers
+{
+    public class STM32F4_Watchdog : BasicDoubleWordPeripheral, IKnownSize
+    {
+        //TODO: Could propably get osc_frequency from RCC.
+        //TODO: Stop timer on debug stop.
+        //TODO: Use RCC to set restart cause.
```

Comment:
regarding your todos:

- rcc is not really properly/robustly implemented. The constructor parameter for frequency is ok.
- debugging with GDB will stop this timer
- can you elaborate on RCC and restart cause?

Reply:
I think it doesn't stop on debug pause. When stm32 resets RCC stores the reason for reset (Software,watchdog etc)